src/Illuminate/Database/Query/Builder.php

```diff
@@ -1427,31 +1441,33 @@ public function simplePaginate($perPage = 15, $columns = ['*'], $pageName = 'pag
      */
     public function getCountForPagination($columns = ['*'])
     {
-        $this->backupFieldsForCount();
+        $selectSql = $this->toSql();
```

Comment:
I think the original line:

```php
$grouped = isset($this->groups) && $selectSql = $this->toSql();
```

Did what you wanted, cause it will not Assign $selectSql if isset($this->groups) is false :smile: